### PR TITLE
Adds the ability to pass a conversion object like JSON

### LINKED ISSUE #11: Make it possible to pass a conversion function to fromAttribute
For attributes that need to be parsed before setting the property, it would be convenient if we could pass an option to `fromAttribute` for how to do this conversion. Also, if we do two-way binding it would be useful to be able to specify how to stringify a property before setting an attribute.

It might be simplest to allow passing something that has a `stringify` and `parse` method so that you could do something like this:

```js
person: { type: Object, bind: fromAttribute( JSON ) }
```

### PR DESCRIPTION
**Pull request by cherifGsoul**
Fixes #11 

Make it possible to pass a JSON like conversion object:

`person: { type: Object, bind: fromAttribute( JSON ) }`

or:

`person: { type: Object, bind: fromAttribute( 'attr-name',  JSON ) }`

### CODE REVIEW

File: from-attribute.js [hunk 1]
```diff
@@ -24,7 +32,12 @@ function initializeFromAttribute (propertyName, ctr, attributeName) {
 	if (ctr[metaSymbol]._attributeChangedCallbackHandler === undefined) {
 		ctr[metaSymbol]._attributeChangedCallbackHandler = {};
 	}
-	if (attributeName === undefined) {
+	if (typeof attributeName === 'object') {
+		if (isJSONLike(attributeName)) {
```
Comment (phillipskevin) [inline from-attribute.js]: Can we give a warning when the wrong thing is passed? This can happen in a future PR.

File: from-attribute-test.js [hunk 2]
```diff
@@ -168,4 +168,64 @@ if (browserSupports.customElements) {
...
+		const bindings = [];
+
+		class MyEl extends HTMLElement {
+			static get observedAttributes() {
+				return attributes;
+			}
+		}
+
+		attributes.forEach(attribute => {
+			const makeFromAttribute = fromAttribute(attribute, JSON);
```
Comment (phillipskevin) [inline from-attribute-test.js:213]: We should make sure that this works correctly if `attribute` is different from the attribute passed in `setAttribute`.

**Review by phillipskevin — CHANGES_REQUESTED** [review]
This looks good. I just had one small comment. Also, we'll need to document these uses, but we can do that in a future PR if you want.

**Comment by cherifGsoul** [comment]
@phillipskevin I updated the PR as requested, can you review it again, please?

**Review by phillipskevin — APPROVED** [review]

**Pull request merged**